src/index.ts

```diff
@@ -336,37 +336,32 @@ export function register (opts: RegisterOptions = {}): Register {
 /**
  * Create TypeScript compiler instance.
  */
-export function create (options: CreateOptions = {}): Register {
-  const optionsWithoutDefaults = options
-  options = defaults(DEFAULTS, options)
-
-  // Require the TypeScript compiler and configuration.
-
+export function create (rawOptions: CreateOptions = {}): Register {
+  const inputOptions = defaults(DEFAULTS, rawOptions)
   const cwd = options.dir ? resolve(options.dir) : process.cwd()
-  const compilerBefore = options.compiler
 
   /**
-   * Load the typescript compiler.  It is required to load the tsconfig but might
+   * Load the typescript compiler. It is required to load the tsconfig but might
    * be changed by the tsconfig, so we sometimes have to do this twice.
    */
-  function loadCompiler () {
-    const compiler = require.resolve(options.compiler || 'typescript', { paths: [cwd, __dirname] })
+  function loadCompiler (name: string) {
+    const compiler = require.resolve(name || 'typescript', { paths: [cwd, __dirname] })
     const ts: typeof _ts = require(compiler)
     return { compiler, ts }
   }
 
-  // compute enough options to read the config file
-  let { compiler, ts } = loadCompiler()
+  // Compute minimum options to read the config file.
+  let { compiler, ts } = loadCompiler(inputOptions.compiler)
 
-  // Read config file
-  const { config, options: tsconfigOptions } = readConfig(cwd, ts, optionsWithoutDefaults)
```

Comment:
Looks like the usage of `optionsWithoutDefaults` here is what broke https://github.com/TypeStrong/ts-node/issues/938. Fixing and adding a test now.